.github/workflows/openms_ci_matrix_full.yml

```
@@ -553,10 +559,48 @@ jobs:
           SIGNING_EMAIL: "apple@openms.de"
+          # Developer ID Installer certificate identity for signing .pkg (public info, not a secret)
           CPACK_PRODUCTBUILD_IDENTITY_NAME: "Developer ID Installer: OpenMS Inc. (C64UCGJ5PL)"
 
+    # Notarize macOS packages for Gatekeeper approval
+    # This step submits the signed package to Apple for notarization and staples the ticket
+    - name: Notarize macOS Package
+      if: steps.set-vars.outputs.pkg_type == 'pkg' && startsWith(matrix.os, 'macos')
+      shell: bash
+      env:
+        CODESIGNPW: ${{ secrets.APPLE_NOTARIZATION_PASSWORD }}
+      run: |
```

actionlint (1.7.9)

shellcheck reported issue in this script: SC2012:info:5:12: Use find instead of ls to better handle non-alphanumeric filenames

(shellcheck)